### DISCUSSION TITLE
Rule that allows anyone to count number of rows without returning field values

### DISCUSSION THREAD

**Discussion by AgeOfMarcus** [pocketbase/pocketbase#2449] [Q&A] [answered]
Is it possible to create a rule for the `List/Search` section that
will allow anyone to query the number of rows matching a certain
filter, but prevent the data (field values) from being viewed? 

I have tried `@request.data.expand = []` but that shows as invalid. Is this possible?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no such rule but as an alternative you can create a [View collection](https://pocketbase.io/docs/collections/#view-collection) (_Admin UI > + New collection > "Type: View"_) with a plain `COUNT`, something like:

```sql
SELECT 1 as id, COUNT(*) as total
FROM `yourCollectionName`
```

View collections are read-only and have their own List and View API rules that you can adjust to restrict the access to the count.

**Reply by AgeOfMarcus** [reply]
Thank you very much. I didn't realize that was a thing. Also sorry for the multiple posts, I didn't even realize this posted successfully!
